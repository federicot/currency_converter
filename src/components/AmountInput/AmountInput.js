import React from 'react';
import { 
  View,
  StyleSheet,
  TextInput
} from 'react-native';
import { Icon } from 'native-base';
import PropTypes from 'prop-types';

const propTypes = {
  currencyCode: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  editable: PropTypes.bool,
  onChange: PropTypes.func
};

const defaultProps = {
  editable: true,
  onChange: () => {}
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
  },
  textInput: {
    flex: 1,
    fontSize: 30,
    textAlign: 'right',
  }
});

const AmountInput = ({ value, editable, onChange }) => {
  return (
    <View style={styles.container}>
      <Icon active type="FontAwesome" name="usd" />
      <TextInput
        editable={editable}
        selectTextOnFocus
        value={value}
        keyboardType="numeric"
        style={styles.textInput}
        onChangeText={onChange}
      />
    </View>
  );
}

AmountInput.propTypes = propTypes;
AmountInput.defaultProps = defaultProps;

export default AmountInput;