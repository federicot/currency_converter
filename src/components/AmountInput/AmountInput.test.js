/* eslint-env jest */
import React from 'react';
import AmountInput from './AmountInput';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <AmountInput
      currencyCode="USD"
      value="10"
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});