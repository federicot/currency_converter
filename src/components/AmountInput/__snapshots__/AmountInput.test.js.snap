// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    active={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#000",
            "fontSize": 30,
          },
        ],
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    type="FontAwesome"
    virtual={undefined}
  >
    ï…•
  </Text>
  <TextInput
    allowFontScaling={true}
    editable={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    selectTextOnFocus={true}
    style={
      Object {
        "flex": 1,
        "fontSize": 30,
        "textAlign": "right",
      }
    }
    value="10"
  />
</View>
`;
